### ISSUE TITLE
Related Courses is a Free Text box in Administrate - 

### ISSUE THREAD

**Issue by JazHaK** [JackHaK/TMF#21]
do we want to get more structured on the web site ??? Should we use Categories?

**Comment by seanthepottingshed** [comment]
@joshgosselin 

What are your thoughts on the above? As it stands the website is expecting a series to course ids to populate related courses.

**Comment by joshgosselin** [comment]
@seanthepottingshed are we setting a specific value of related courses (2, 3,5 ???)

My thoughts are that we can default to courses from the same category. 
But maybe with the functionality that they can define a specific course if appropriate. 

Not sure what your thoughts are for the best implementation of this. 

Could we allow the GTA to enter the ID into the free text field in administrate and parse that in the integration tier. And populating any remaining slots with courses based on category. Obviously they need to get the CSV ids structure correct, but does allow control from Administrate.

**Comment by seanthepottingshed** [comment]
@joshgosselin 

We aren't setting a specific value of **Related Courses,** each **Course** when imported will have a reference to the `integration_tier_id`, the relationship to **Related Courses** would be via `integration_tier_id`'s.

It's easy enough to amend the current logic to fetch 5 random courses from the categories that the course belongs to on our side ( we'd previously agreed that 5 should be the limit for related courses displayed in a course details page ), and as you've suggested allow other `integration_tier_id`'s to be entered via **Adminstrate** and populate the remaining 5 as before.

**Comment by joshgosselin** [comment]
Yep make sense to it all to be managed by the integration tier. 

@JackHaK does this make sense to you: 
 - Parse CSV IDs from Administrate
 - Potentially build a picker UI in integration tier in the future to manage/debug if automated import isn't working
- if more then 5 IDs are entered provide us with a random 5
- if less then 5 IDs populate up to 5 with courses with matching category.

**Comment by PeteHaK** [comment]
The integration tier uses Administrate IDs so this will work – and actually the Administrarte screens use Course IDs to navigate so using course ids in the related courses fields comma delimited will work fine.

 

We’re gonna struggle to support this channel today as we’re all busy but queue up the issues and we’ll turn them around overnight if we can.

 

Pete

**Comment by seanthepottingshed** [comment]
@PeteHaK 

Thanks, appreciated.

**Comment by joshgosselin** [comment]
No worries @PeteHaK, we have plenty to crack on with so as and whenever works for you.

**Comment by PeteHaK** [comment]
relatedCourses added as a simple JSON string array of course IDs. The administrate field adds HTML tags which we've removed on inflate - there is a chance that someone putting rubbish in Administrate will cause us an issue - lets see

**Comment by seanthepottingshed** [comment]
@PeteHaK @joshgosselin 

Thanks for sorting out the `relatedCourses` this now works:

* https://gta.tps.digital/courses/finance/2018-compliance-cpd-series/646

Is it possible to ensure that there are always 5 `relatedCourses` as per @joshgosselin [comment above](https://github.com/JackHaK/TMF/issues/21#issuecomment-429775345)?

* if more then 5 IDs are entered provide us with a random 5
* if less then 5 IDs populate up to 5 with courses with matching category.

**Comment by seanthepottingshed** [comment]
@PeteHaK @JazHaK @JackHaK @joshgosselin 

During the process of updating API integration, I've noticed that the following:

* http://gtadev.humbleandkind.co.uk/api/courses/646

...has related course ids of 

```php
[302, 303, 304, 305]
```

However there is no course with an id of `304`:

* http://gtadev.humbleandkind.co.uk/api/courses/304

**Comment by PeteHaK** [comment]
That was just test data – sorry. Unfortunately the data comes from Administrate as free text so there is no way we can validate the input. 

 

There is a chance that GTA will do this so you will need to cater for IDs that don’t exist. Alternatively we could mark invalid courses on the way through the integration tier if easier.

**Comment by seanthepottingshed** [comment]
@PeteHaK 

No worries, thanks for the clarification.

**Comment by seanthepottingshed** [comment]
I will continue to ignore related events without ids.
